MSAL/src/public/MSALError.h

```diff
@@ -117,235 +115,223 @@ extern NSString *MSALBrokerVersionKey;
-            </dict>
-
-     */
-    MSALErrorRedirectSchemeNotRegistered    = -42001,
+extern NSString *MSALErrorDomain;
 
-    MSALErrorInvalidRequest                 = -42002,
-    MSALErrorInvalidClient                  = -42003,
-    MSALErrorInvalidGrant                   = -42004,
-    MSALErrorInvalidScope                   = -42005,
-    MSALErrorUnauthorizedClient             = -42006,
+// TODO: update error numbers for each enum?
+typedef NS_ENUM(NSInteger, MSALServiceErrorCode)
+{
+    MSALErrorInvalidRequest                     = -42002,
+    MSALErrorInvalidClient                      = -42003,
+    MSALErrorInvalidGrant                       = -42004,
+    MSALErrorInvalidScope                       = -42005,
+    MSALErrorUnauthorizedClient                 = -42006,
     
     /*!
      The server returned an unexpected http response. For instance, this code
      is returned for 5xx server response when something has gone wrong on the server but the
      server could not be more specific on what the exact problem is.
      */
-    MSALErrorUnhandledResponse              = -42007,
-
-    MSALErrorServerDeclinedScopes           = -42008,
-
-    /*! 
-        The passed in authority URL does not pass validation.
-        If you're trying to use B2C, you must disable authority validation by
-        setting validateAuthority of MSALPublicClientApplication to NO.
-     */
-    MSALErrorFailedAuthorityValidation = -42010,
+    MSALErrorUnhandledResponse                  = -42007,
     
     /*!
-        Interaction required errors occur because of a wide variety of errors
-        returned by the authentication service. In all cases the proper response
-        is to use a MSAL interactive AcquireToken call with the same parameters.
-        For more details check MSALOAuthErrorKey and MSALOAuthErrorDescriptionKey
-        in the userInfo dictionary.
+     The user or application failed to authenticate in the interactive flow.
+     Inspect MSALOAuthErrorKey and MSALErrorDescriptionKey in the userInfo
+     dictionary for more detailed information about the specific error.
      */
-    MSALErrorInteractionRequired        = -42100,
-    MSALErrorMismatchedUser             = -42101,
-    MSALErrorNoAuthorizationResponse    = -42102,
-    MSALErrorBadAuthorizationResponse   = -42103,
+    MSALErrorAuthorizationFailed                = -42104,
     
     /*!
-        The user or application failed to authenticate in the interactive flow.
-        Inspect MSALOAuthErrorKey and MSALErrorDescriptionKey in the userInfo
-        dictionary for more detailed information about the specific error.
+     Response was received in a network call, but the response body was invalid.
+     
+     e.g. Response was to be expected a key-value pair with "key1" and
+     the json response does not contain "key1" elements
+     
      */
-    MSALErrorAuthorizationFailed = -42104,
+    MSALErrorInvalidResponse                    = -42600,
```

Comment:
what's the difference between this one and MSALErrorUnhandledResponse?

Reply:
MSALErrorUnhandledResponse is for http errors, MSALErrorInvalidResponse is for response parsing error.